pkg/microservice/aslan/core/common/repository/mongodb/task.go

```diff
@@ -380,6 +389,35 @@ func (c *TaskColl) List(option *ListTaskOption) (ret []*TaskPreview, err error)
 	return
 }
 
+func (c *TaskColl) ListPreview(pipelineNames []string) (ret []*TaskPreview, err error) {
+	ret = make([]*TaskPreview, 0)
+	query := bson.M{}
+	if pipelineNames != nil {
+		query["pipeline_name"] = bson.M{"$in": pipelineNames}
+	}
+	query["is_archived"] = false
+	query["is_deleted"] = false
+	selector := bson.D{
+		{"task_id", 1},
+		{"task_creator", 1},
+		{"product_name", 1},
+		{"pipeline_name", 1},
+		{"status", 1},
+		{"create_time", 1},
+		{"start_time", 1},
+		{"end_time", 1},
+		{"type", 1},
+	}
+	opt := options.Find()
+	opt.SetProjection(selector)
+	cursor, err := c.Collection.Find(context.TODO(), query, opt)
+	if err != nil {
+		return
```

Comment:
the named return value err has beed redecleared in line 413, so the return value in this `if statement` would be nil,nil